py/src/braintrust/oai.py

```diff
@@ -1,41 +1,128 @@
-# TODO MOVE TO AUTOEVALS IMPORT
-# https://github.com/braintrustdata/braintrust/issues/218
+import time
 
-import json
-import sqlite3
+from .logger import current_span
 
-from .cache import CACHE_PATH
 
-OAI_CACHE = CACHE_PATH / "oai.sqlite"
+class ChatCompletionWrapper:
```

Comment:
I wonder if this wrapper could be used for more than just `ChatCompletion`. What about any object that derives from `EngineAPIResource`?

Reply:
Each one has distinct things we'd want to log I think

Reply:
Is that true? I think both [`Completion`](https://platform.openai.com/docs/api-reference/completions/create) and [`Edit`](https://platform.openai.com/docs/api-reference/edits) have the same schema, as far as we are drilling down.

Reply:
I haven't looked at the return types -- in general i think we should proceed with caution until we thoroughly test.